starport/cmd/module-import.go

```diff
@@ -30,7 +34,12 @@ func importModuleHandler(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	if err := sc.ImportModule(name); err != nil {
+	ctx := WithValidation(context.Background(), cmd)
```

Comment:
I think context.Value is not great for type safety and makes the the software hard to debug. How about making the placeholder package Newable instead?

Reply:
not sure i understand what Newable means. 

you are right that context.Value is not type safe, but I don't think that this is always an issue. 
for instance all the logic of working with relevant value is encapsulated in placeholder package, and I don't see how It can be trivially broken.

Reply:
do you mean something along this lines
```go

func New() Placeholder

func (p *Placeholder) Replace

type Context struct {
    Placeholder 
} 
```

this is what I meant by application-specific context. If you prefer it I can make the change, I thought that it is not worth it until we have a more complex validation pipeline.

Reply:
I beileve that context.Value should be the last option specially because it makes it hard to track how the code flows. I think it is comprehensible to use it for 3rd party packages that does not provide a mechanism to pass data between each layer for example http.Handler.

But here, we got the whole control over scaffolder, and templates packages. We can basically intoruduce a `placeholder.New` func to create an insitance of `Placeholder` and pass it to templates. In this case funcs like `Validate` and `Replace` becomes to a method of `Placeholder` and we keep the data inside `Placeholder`.

Reply:
I'm suggesting to completey dropping use of Context values.

Reply:
with context you don't have to change signature of the function each time when you have a need for a new session specific object

Reply:
for the pattern that i am describing consider the usage of sdk.Context is cosmos-sdk